### PR TITLE
Tools: Add ardupilot_gazebo install script

### PR DESCRIPTION
**Pull request by snktshrma**
Added ardupilot_gazebo package for simulation purposes

### CODE REVIEW

File: Tools/environment_install/install-ROS-ubuntu.sh [hunk 1]
```diff
@@ -246,6 +247,26 @@ else
     echo "Skipping adding ardupilot_ws to your home folder."
 fi
 
+
+if maybe_prompt_user "Add ardupilot_gazebo to your home folder [N/y]?" ; then
+    if [ ! -d $AP_GZ_ROOT ]; then
+        git clone https://github.com/khancyr/ardupilot_gazebo
```
Comment (rmackay9) [inline Tools/environment_install/install-ROS-ubuntu.sh:253]: This should be "https://github.com/ArduPilot/ardupilot_gazebo" I think.

**Review by rmackay9 — CHANGES_REQUESTED** [review]

**Comment by rmackay9** [comment]
Thanks for this!  I've asked @jmachuca77 and @srmainwaring for their review too.

**Review by khancyr — COMMENTED** [review]
this need more than this as the plugin got some dependancies to build

File: Tools/environment_install/install-ROS-ubuntu.sh (hunk 1 above)
Comment (snktshrma) [reply Tools/environment_install/install-ROS-ubuntu.sh]: Made the changes @rmackay9 sir

**Comment by snktshrma** [comment]
@khancyr I have made the required changes and most of the dependencies will be downloaded while downloading and setting ardupilot.

**Review by jmachuca77 — APPROVED** [review]

**Comment by rmackay9** [comment]
Merged, thanks!

File: Tools/environment_install/install-ROS-ubuntu.sh [hunk 2]
```diff
@@ -246,6 +247,27 @@ else
     echo "Skipping adding ardupilot_ws to your home folder."
 fi
 
+
+if maybe_prompt_user "Add ardupilot_gazebo to your home folder [N/y]?" ; then
+    if [ ! -d $AP_GZ_ROOT ]; then
+        sudo apt install libgz-sim7-dev rapidjson-dev
```
Comment (srmainwaring) [inline Tools/environment_install/install-ROS-ubuntu.sh:253]: `sudo apt install gz-garden rapidjson-dev` is needed to install all the libraries. See https://gazebosim.org/docs/garden/install_ubuntu.

File: Tools/environment_install/install-ROS-ubuntu.sh [hunk 3]
```diff
@@ -246,6 +247,27 @@ else
...
+        git clone https://github.com/ArduPilot/ardupilot_gazebo
+        pushd $AP_GZ_ROOT
+        mkdir build && pushd build
```
Comment (srmainwaring) [inline Tools/environment_install/install-ROS-ubuntu.sh:256]: Need to create the workspace directory first, then clone, then cd into the project, then make build dir and build.

```bash
mkdir -p $AP_GZ_ROOT/src
pushd $AP_GZ_ROOT/src
git clone https://github.com/ArduPilot/ardupilot_gazebo
pushd ardupilot_gazebo
mkdir build && pushd build
```

**Comment by srmainwaring** [comment]
@rmackay9 - only had a chance to look at this. There are a few issues with the script I'm afraid. I've started to add notes but hit a couple of snags with the script for Ubuntu 20.04 / ROS Noetic which I'm documenting in https://github.com/ArduPilot/ardupilot/pull/22484.

We may also hit some issues with a desktop-full install of ROS1 which uses catkin and Gazebo11 and the plugin build which targets Gazebo Sim uses colcon. They should co-exist but I'm seeing a couple of package dependency issues for the ROS1 install when I rerun the script after install gz-garden...

**Comment by rmackay9** [comment]
@srmainwaring,

Thanks very much for reviewing this.  Sorry I merged this too quickly.  I don't really understand this all that well so from now on I might hold off on merging PRs in this area until you've had a chance to review.  Thanks again!

**Comment by snktshrma** [comment]
My apologies for these issues! Earlier I was working with Khancyr's and SwiftGust's version of ardupilot_gazebo package and these instructions were working pretty fine!

Anyhow, thanks for these changes @srmainwaring

**Pull request merged**